rampwf/hyperopt/tests/interfaces/header_in_files/titanic/submissions/starting_kit/classifier.py

```diff
@@ -0,0 +1,27 @@
+from sklearn.linear_model import LogisticRegression
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import Pipeline
+from sklearn.base import BaseEstimator
+from rampwf.hyperopt import Hyperparameter
+
+# RAMP START HYPERPARAMETERS
+logreg_C = Hyperparameter(
+    dtype='float', default=1.0, values=[0.01, 0.1, 0.9, 1.0])
+imputer_strategy = Hyperparameter(
+    dtype='object', default='median', values=['mean', 'median'])
```

Comment:
I feel these changes are too invasive for ramp-workflow and it tends to reinvent the notion of parameter space present in hyperopt and scikit-optimize (https://scikit-optimize.github.io/notebooks/hyperparameter-optimization.html). I would do something like this:

```
logreg_C = 1.0  # values : [0.01, 0.1, 0.9, 1.0]
imputer_strategy = 'median'  # values : ['mean', 'median']
```

by doing this hyperopt does not need yet to be in core ramp-workflow.
A first PR here would be to be able to parse the header, eventually
print a warning when not found.

to embrace more the hyperopt or scikit-optimize approach it could be:

```
logreg_C = 1.0  # Real(10**-5, 10**0, "log-uniform")
imputer_strategy = 'median'  # Categorical(['mean', 'median'])
```

Reply:
We discussed the syntax options with Amine and decided to go with the pure python declaration for two reasons: 1) the python interpreter checks syntax instead of our parser and 2) it was easier to get the hypers by loading the file and filtering the Hyperparameter objects. But I don't have a strong opinion, if someone feels strongly enough to implement the comment option, we could add both and see which one is used.

On the  scikit-optimize/hyperopt way of defining spaces I have a stronger opinion, it's an overly complicated design. No reason to have continuous spaces, in my experience it's the data scientist's prior knowledge what values to try, but tell me the use case in which you really need continuous (ie you have no idea about the granularity/smoothness of the score wrt your hyper) and I change my mind. In this way you don't need distribution parameter either, just specify a logarithmic grid if you want. We also have an optional `prior` in the declaration that you can communicate to your hyperopt engine (random uses it already). Finally, another advantage is that we can use bandits, and even GP is much happier on a predefined grid (e.g. you can measure nuggets/noise easily).